library/modules/Military.cpp

```diff
@@ -389,6 +394,19 @@ static void remove_officer_entity_link(df::historical_figure* hf, df::squad* squ
     former_pos->link_strength = 100;
 
     hf->entity_links.push_back(former_pos);
+
+    int32_t event_id = *df::global::hist_event_next_id;
```

Comment:
```suggestion
    int32_t event_id = (*df::global::hist_event_next_id)++;
```

Reply:
this should work, allowing you to remove the explicit increment below

Reply:
Yep, I tested that earlier before publishing this but could not tell which would be more readable. if you prefer it I will do just that.